### PR TITLE
add missing arguments for all constructors of ListView and GridView

### PR DESCRIPTION
**Pull request by zmtzawqlp**
## Description

dragStartBehavior  and keyboardDismissBehavior arguments are not available for all constructors of ListView and GridView.

## Tests

 I added the following tests:

'GridView dismiss keyboard onDrag test'
'GridView dismiss keyboard manual test'

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by Hixie** [comment]
Would be good to add tests that verify that these constructor arguments end up setting the right properties, just so that nobody "simplifies" the code by removing them (which right now would not make any tests fail).

Also looks like you have some trailing spaces somewhere. See the analyze-linux bot's logs.

Other than that, seems good!

**Comment by zmtzawqlp** [comment]
> Would be good to add tests that verify that these constructor arguments end up setting the right properties, just so that nobody "simplifies" the code by removing them (which right now would not make any tests fail).
> 
> Also looks like you have some trailing spaces somewhere. See the analyze-linux bot's logs.
> 
> Other than that, seems good!

thanks, I have finished them, please take a look when you have a time.

File: packages/flutter/test/widgets/scroll_view_test.dart [hunk 1]
```diff
@@ -128,6 +128,38 @@ void main() {
...
+          height: 50,
+          color: Colors.green,
+          child: TextField(
+              focusNode: focusNodes[index],
+              style: const TextStyle(
+                fontSize: 24,
+                fontWeight: FontWeight.bold,
+              )));
+      },
+    )));
```
Comment (goderbauer) [inline packages/flutter/test/widgets/scroll_view_test.dart:151]: nit: fix indentation:

```suggestion
      child: GridView.builder(
      padding: const EdgeInsets.all(0),
      gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount:2),
      keyboardDismissBehavior: ScrollViewKeyboardDismissBehavior.onDrag,
      itemCount: focusNodes.length,
      itemBuilder: (BuildContext context, int index){
        return Container(
          height: 50,
          color: Colors.green,
          child: TextField(
              focusNode: focusNodes[index],
              style: const TextStyle(
                fontSize: 24,
                fontWeight: FontWeight.bold,
              ),
            ),
          );
        },
      ),
    ));
```

File: packages/flutter/test/widgets/scroll_view_test.dart [hunk 2]
```diff
@@ -159,6 +191,38 @@ void main() {
     expect(textField.focusNode.hasFocus, isTrue);
   });
 
+  testWidgets('GridView dismiss keyboard manual test', (WidgetTester tester) async {
+    final List<FocusNode> focusNodes = List<FocusNode>.generate(50, (int i) => FocusNode());
+
+    await tester.pumpWidget(textFieldBoilerplate(
+        child: GridView.builder(
```
Comment (goderbauer) [inline packages/flutter/test/widgets/scroll_view_test.dart:198]: same here

File: packages/flutter/test/widgets/scroll_view_test.dart (hunk 1 above)
Comment (zmtzawqlp) [reply packages/flutter/test/widgets/scroll_view_test.dart:151]: thanks for your review. I have some doubt about format.  I have tried to format file  by  flutter format xxx.dart ,  but it doesn't meet the requirements. Could you please tell me how to check it and what's the standard? thanks

Comment (zmtzawqlp) [reply packages/flutter/test/widgets/scroll_view_test.dart:151]: @Hixie Could you please clarify the rule of flutter format when you have a time? i have tried flutter format xxx.dart and   flutter analyze. but it seems not good.

Comment (goderbauer) [reply packages/flutter/test/widgets/scroll_view_test.dart:151]: The rules for the style of this repository are written down in https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo.

Most notably, this repository does not use automatic formatting (e.g. with dartformat or `flutter format`). This rule about trailing commas is also relevant for this particular case: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use-a-trailing-comma-for-arguments-parameters-and-list-items-but-only-if-they-each-have-their-own-line

Comment (zmtzawqlp) [reply packages/flutter/test/widgets/scroll_view_test.dart]: It's kind of you. Thank you for your patience. please take a look when you have a time.

File: packages/flutter/test/widgets/scroll_view_test.dart (hunk 2 above)
Comment (zmtzawqlp) [reply packages/flutter/test/widgets/scroll_view_test.dart]: done

File: packages/flutter/test/widgets/scroll_view_test.dart [hunk 3]
```diff
@@ -128,6 +128,41 @@ void main() {
     expect(textField.focusNode.hasFocus, isFalse);
   });
 
+  testWidgets('GridView dismiss keyboard onDrag test', (WidgetTester tester) async {
```
Comment (chunhtai) [inline packages/flutter/test/widgets/scroll_view_test.dart:131]: We need test for each constructor as well

**Review by chunhtai — COMMENTED** [review]
Change is reasonable, however, we need more test coverage

**Review by goderbauer — APPROVED** [review]
LGTM modulo the comment above.

File: packages/flutter/test/widgets/scroll_view_test.dart (hunk 3 above)
Comment (zmtzawqlp) [reply packages/flutter/test/widgets/scroll_view_test.dart:131]: As I see,  arguments are refer to BoxScrollView,  is it a duplicate test code for that?

**Review by chunhtai — APPROVED** [review]
LGTM

**Pull request merged**